File: app/src/main/java/com/kent/university/privelt/ui/data/DataViewHolder.java
package com.kent.university.privelt.ui.data;

import android.text.TextUtils;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.kent.university.privelt.R;
import com.kent.university.privelt.model.UserData;
import com.university.kent.dataextractor.model.UserDataTypes;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import butterknife.BindView;
import butterknife.ButterKnife;

public class DataViewHolder extends RecyclerView.ViewHolder {

    @BindView(R.id.title)
    TextView title;

    @BindView(R.id.value)
    TextView value;

    @BindView(R.id.concatenated_data)
    TextView concatenatedData;

    @BindView(R.id.icon)
    ImageView icon;

    DataViewHolder(@NonNull View itemView) {
        super(itemView);

        ButterKnife.bind(this, itemView);
    }

    public void bind(UserData userData) {
        title.setText(userData.getType());
        value.setText(userData.getValue());
        concatenatedData.setText(TextUtils.join("\n", userData.getUnConcatenatedData()));

        UserDataTypes userDataType = UserDataTypes.valueOf(userData.getType().toUpperCase());
        icon.setImageResource(userDataType.getRes());
    }
}

File: app/src/main/java/com/kent/university/privelt/ui/data/DataViewHolder.kt
/*
 *  This Source Code Form is subject to the terms of the Mozilla Public
 *  * License, v. 2.0. If a copy of the MPL was not distributed with this
 *  * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */
package com.kent.university.privelt.ui.data

import android.text.TextUtils
import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.kent.university.privelt.model.UserData
import com.kent.university.privelt.utils.sentence.SentenceAdapter
import net.neferett.webviewsextractor.model.UserDataTypes
import java.util.*
import kotlinx.android.synthetic.main.cell_userdata.view.*

class DataViewHolder internal constructor(itemView: View) : RecyclerView.ViewHolder(itemView) {

    fun bind(userData: UserData) {
        itemView.title!!.text = SentenceAdapter.capitaliseFirstLetter(userData.type)
        itemView.value!!.text = userData.value
        itemView.concatenated_data!!.text = TextUtils.join("\n", userData.unConcatenatedData)
        val userDataType = UserDataTypes.valueOf(userData.type.toUpperCase(Locale.ROOT))
        itemView.icon!!.setImageResource(userDataType.res)
    }
}